spidermon/contrib/scrapy/monitors.py

```diff
@@ -87,23 +88,60 @@ def test_should_finish_with_expected_reason(self):
 class UnwantedHTTPCodesMonitor(BaseScrapyMonitor):
     """Check for maximum number of unwanted HTTP codes.
 
-    You can configure a ``dict`` of unwanted HTTP codes with
+    Usage # 1
+
+    You can configure an ``int`` of maximum unwanted HTTP codes with
+    ``SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT`` the default value is::
+
+        SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT = 10
+
+        ----->
+
+        SPIDERMON_UNWANTED_HTTP_CODES = {
+            code: SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT
+            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+        }
+
+    Usage # 2
+
+    You can configure a ``dict`` or ``list`` of unwanted HTTP codes with
     ``SPIDERMON_UNWANTED_HTTP_CODES`` the default value is::
 
-        DEFAULT_ERROR_CODES = {
+        1.1 - In case of Dictionary:
+
+        SPIDERMON_UNWANTED_HTTP_CODES = {
             code: 10
-            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]}
-    """
+            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+        }
+
+        1.2 - In case of List:
 
-    DEFAULT_ERROR_CODES = {
-        code: 10 for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
-    }
+        SPIDERMON_UNWANTED_HTTP_CODES = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+
+        ----->
+
+        SPIDERMON_UNWANTED_HTTP_CODES = {
+            code: SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT / 10
+            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+        }
+
+    """
+    DEFAULT_ERROR_MAX_COUNT = 10
+    DEFAULT_ERROR_CODES = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
 
     @monitors.name("Should not hit the limit of unwanted http status")
     def test_check_unwanted_http_codes(self):
-        error_codes = self.crawler.settings.getdict(
```

Comment:
Instead of `error_codes` I would use `unwanted_http_codes` which shows more clearly what we are trying to verify in this monitor (not all codes are errors). Maybe change it in the whole monitor would be good.

Reply:
Done